lib/iris/tests/test_concatenate.py

```diff
@@ -748,6 +730,49 @@ def test_concat_2x2d_aux_xy_bounds(self):
         self.assertEqual(len(result), 1)
         self.assertEqual(result[0].shape, (2, 4))
 
+    def test_fill_value_invariant_to_order__same_non_None(self):
+        y = (0, 2)
+        fill_value = 1234
+        cubes = [_make_cube((i*2, i*2+2), y, i, mask=True,
+                            fill_value=fill_value) for i in range(3)]
+        for combo in itertools.permutations(cubes):
+            result = iris.cube.CubeList(combo).concatenate_cube()
+            self.assertEqual(result.fill_value, fill_value)
+            self.assertEqual(result.data.fill_value, fill_value)
+
+    def test_fill_value_invariant_to_order__all_None(self):
+        y = (0, 2)
+        cubes = [_make_cube((i*2, i*2+2), y, i, mask=True,
+                            fill_value=None) for i in range(3)]
+        for combo in itertools.permutations(cubes):
+            result = iris.cube.CubeList(combo).concatenate_cube()
+            self.assertIsNone(result.fill_value)
+            np_fill_value = ma.masked_array(0, dtype=result.dtype).fill_value
+            self.assertEqual(result.data.fill_value, np_fill_value)
+
+    def test_fill_value_invariant_to_order__different_non_None(self):
```

Comment:
Other than testing the different order, is this much different to [test_concat_masked_2y2d__default_fill_value_from_diff](https://github.com/SciTools/iris/pull/2520/files#diff-af298a2170476a32abb0c94b312b4ff5R438)?

I don't understand what you gain from doing all the permutations?